underfs/s3/src/main/java/alluxio/underfs/s3/S3UnderFileSystemFactory.java

```diff
@@ -39,20 +39,23 @@
   public UnderFileSystem create(String path, Configuration configuration, Object unusedConf) {
     Preconditions.checkNotNull(path);
     Preconditions.checkNotNull(configuration);
+    AWSCredentials creds;
 
     if (addAndCheckAWSCredentials(configuration)) {
-      AlluxioURI uri = new AlluxioURI(path);
-      try {
-        return new S3UnderFileSystem(uri.getHost(), configuration);
-      } catch (ServiceException e) {
-        LOG.error("Failed to create S3UnderFileSystem.", e);
-        throw Throwables.propagate(e);
-      }
+      creds = new AWSCredentials(configuration.get(Constants.S3_ACCESS_KEY),
+              configuration.get(Constants.S3_SECRET_KEY));
+    } else {
+      creds = AWSEC2IAMSessionCredentials.loadFromEC2InstanceData(true);
```

Comment:
What would happen if the AWS credentials are not specified, and the application is NOT running on EC2? It seems like there is one (error) case not handled here?